src/portable/espressif/esp32s2/dcd_esp32s2.c

```diff
@@ -167,15 +169,25 @@ void dcd_init(uint8_t rhport)
 {
   ESP_LOGV(TAG, "DCD init - Start");
 
-  // A. Disconnect
-  ESP_LOGV(TAG, "DCD init - Soft DISCONNECT and Setting up");
-  USB0.dctl |= USB_SFTDISCON_M; // Soft disconnect
+  //Detect if USB Persistence flag is set
+  //If flag is set, then we should assume that the USB periperal is already running,
+  //that we have not changed any descriptors, so we can skip enumeration and continue from there
+  bool did_persist = (USB_WRAP.date.val & USBDC_PERSIST_ENA) != 0;
```

Comment:
any reason for testing with bitmask here and in the bsp files it does equality test against USBDC_PERSIST_ENA?

Reply:
hold your horses @atanisoft :) code is not final.

Reply:
Yeah, I know it's not final but it is curious why the two approaches and if it might be playing a factor in the inconsistency you are seeing.

Reply:
@atanisoft ... why are you trying to find reasons where there are none? I Ctrl+Z some code, which brought the old approach that you had issue with before... this bit that is read by the bootloader can in no way ever cause the problem I am describing. and it's the same exact thing anyway...

Reply:
It's just a Q on why the inconsistent usage of `USBDC_PERSIST_ENA` between the bsp files and this one. There is no technical issue between the two approaches that I can think of but more of a style issue.

Reply:
maybe wait for the WIP to be removed an judge then :)

Reply:
Let's focus on the actual functionality and less on the style. Those aren't too important for new feature PR, I am fine with any ugly/hacking code as long as it makes sense and doesn't break thing. It is much easier to refactor a working code later on. Thanks.